hr_timesheet_task_required/models/account_analytic_line.py

```diff
@@ -17,11 +17,10 @@ class AccountAnalyticLine(models.Model):
     @api.constrains(
         'project_id',
         'task_id',
-        'is_task_required',
     )
     def _check_timesheet_task(self):
         for line in self:
-            if line.project_id and line.is_task_required and not line.task_id:
+            if line.is_task_required and line.project_id and not line.task_id:
```

Comment:
```suggestion
            if line.is_task_required and not line.task_id:
```

Reply:
@pedrobaeza please elaborate, not sure I follow why we should check projectless AALs?

Reply:
(that's not timesheet entry by design, AFAIK)

Reply:
Because it no project specified (which is not possible any way, as it's required), `line.is_task_required` will be `False`.

Reply:
Ah, true. Great catch! Updated